File: solutions/zig/bob/1/bob.zig
const std = @import("std");

pub fn response(s: []const u8) []const u8 {
    const trimmed = std.mem.trim(u8, s, &std.ascii.whitespace);
    var hasLower = false;
    var hasUpper = false;
    var lastChar: u8 = ' ';
    if (trimmed.len == 0) {
        return "Fine. Be that way!";
    }
    for (trimmed) |char| {
        if (char >= 'a' and char <= 'z') {
            hasLower = true;
        } else if (char >= 'A' and char <= 'Z') {
            hasUpper = true;
        }
        lastChar = char;
    }
    if (!hasLower and hasUpper) {
        if (lastChar == '?') {
            return "Calm down, I know what I'm doing!";
        }
        return "Whoa, chill out!";
    }
    if (lastChar == '?') {
        return "Sure.";
    }
    return "Whatever.";
}

File: solutions/go/bob/1/bob.go
package bob

import "strings"

func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	if remark == "" {
		return "Fine. Be that way!"
	}

	hasLower := false
	hasUpper := false
	var lastChar rune
	for _, c := range remark {
		if c >= 'a' && c <= 'z' {
			hasLower = true
		}
		if c >= 'A' && c <= 'Z' {
			hasUpper = true
		}
		lastChar = c
	}
	if !hasLower && hasUpper {
		if lastChar == '?' {
			return "Calm down, I know what I'm doing!"
		}
		return "Whoa, chill out!"
	}
	if lastChar == '?' {
		return "Sure."
	}
	return "Whatever."
}
